lib/bank_accounts/account.rb

```diff
@@ -0,0 +1,125 @@
+# throw error it's more of a prog thing like a data type that won't be recognized by a method.
+# if it's more the user trying to take too much, return false b/c the method call is where the error message should trigger.
+
+# be consistent about throwing errors vs false.
+
+# PRIVATE might be the cause / PROTECTED might be the solution
+# nope not returning true/false
+# !R private vs protected
+
+
+module BankAccounts
+  class Account
+    ##--------------------------------------------------------------------------
+    # CONSTANTS
+
+
+    # limits
+    MINIMUM_BALANCE = 0
+
+
+    ##--------------------------------------------------------------------------
+    # INSTANCE METHODS
+
+
+    attr_reader :balance, :id
+
+
+    def initialize(id, initial_balance)
+      if initial_balance >= MINIMUM_BALANCE
+        @balance = 0
+        update_balance(initial_balance.round(2))
+      else
+        raise ArgumentError.new("An account cannot be created with a negative balance.")
+      end
+
+      if (@balance >= RECOMMEND_MONEY_MARKET) && (@balance < RECOMMEND_SISTER_BANK) && (self.class != MoneyMarketAccount)
+        puts MONEY_MARKET_RECOMMENDATION
+      end
+
+      if (@balance >= RECOMMEND_SISTER_BANK)
+        raise ArgumentError.new(SISTER_BANK_RECOMMENDATION)
+      end
+
```

Comment:
Similar comment here: you could add the line `raise ArgumentError.new(SISTER_BANK_RECOMMENDATION) if initial_balance >= RECOMMEND_SISTER_BANK` to the top of the `initialize` method, and then you could get rid of the `if` block that raises it, and also get rid of the middle conditional (`@balance < RECOMMEND_SISTER_BANK`) in the previous `if` block, since the method would already have raised an error if that condition wasn't met. It can get hard to read a series of a lot of conditionals, so it's good to simplify them when you can.